### PR TITLE
Remove 404 page!

### PR DESCRIPTION
**Pull request by jenniferexong**
Removing this 404 page - as part of deploying new 404 from static site

[vimaly story](https://vimaly.com/#j8mqm/t/www_Remove_the_old_Middleman_404_page_so_we_see_th.../1PgiM4LNfZrifGHZk)

### CODE REVIEW

**Comment by jenniferexong** [comment]
Seems to have worked! https://staging-www.sharesight.com/nz/404

File: spec/features/robots_spec.rb [hunk 1]
```diff
@@ -8,7 +8,6 @@
   it "should have a sitemap index" do
     expect(page).to have_text('User-agent: *')
     expect(page).to have_text('Disallow: /survey-thanks')
-    expect(page).to have_text('Disallow: /404')
```
Comment (kylorhall) [inline spec/features/robots_spec.rb:11]: Let's keep this in here…I think we should still disallow it (even if it doesn't exist in this repo).

Comment (kylorhall) [reply spec/features/robots_spec.rb:11]: Can you just hardcode this into the `robots.txt.erb` I think?

```rb
<%# Disallow the 404 page in Gatsby (TODO: Migrate this to Gatsby when that owns the robots.txt) %>
Disallow: /404
```

**Review by kylorhall — CHANGES_REQUESTED** [review]

File: spec/features/robots_spec.rb (hunk 1 above)
Comment (kylorhall) [reply spec/features/robots_spec.rb:11]: Actually…do we need to disallow all `/nz/404` too?  I can write that up one sec…

Comment (kylorhall) [reply spec/features/robots_spec.rb:11]: All resolved if CI passes!

**Review by kylorhall — APPROVED** [review]
LGTM assuming CI passes, still run it by QA.

**Pull request merged**